### ISSUE TITLE
Any tips to speed things up?

### ISSUE THREAD

**Issue by Julianoe** [lowfuel/progrockdiffusion#69]
My config is  an AMD Ryzen 5 3600X 6-Core @ 12x 3.8GHz and an AMD Radeon RX 5600 XT with 16GB of RAM (i'm on linux Manjaro).
The generation of a single image is announced to take ~3h30. Do you have any tips to speed things up? I read when launching the command that it uses the CPU. I know it's not an nvidia but wouldn't it be a bit faster if I used the GPU? I don't seem to manage to make it try that. Is that because it only works with Nvidia?

Just out of curiosity what are the time range you guys manage to generate an image with this?

Thanks :)

**Comment by lowfuel** [comment]
On an NVIDIA gpu the default settings image will take about 5 to 10 minutes.

As far as I am aware there is no definitive way yet to run it on an AMD GPU. They have very limited support for Pytorch, and only a few cards that are enabled to work even within that limited environment, sadly.

I don't have an AMD card myself so I can't do anything here to get it working. But if you want to jump into that rabbit hole you're welcome to try! Start by looking at Pytorch's site for the ROCm version, and be ready to do a lot of googling to try and solve whatever error messages come up when you try to run it. I'll help as much as I can from here, but without an AMD card there's not a ton I can do, sadly.

For what it's worth, I briefly tried to get it working on my Steamdeck when I had it, since that technically has an RDNA2 GPU and linux, but I could not figure it out.

**Comment by Julianoe** [comment]
Thanks very much for those guiding thoughts. I don't know if I'll manage to make it work but with the knowledge of Docker and advanced googling, I might try. The ROCm/pytorch info was what I needed to start to search for solutions.

5 to 10 minutes would be a lot more reasonable to try to tinker with this amazing tech at home. 
Thanks again! I'll post here if I manage to make anything work faster on my setup for others to use later.

**Comment by lowfuel** [comment]
Sounds good. Keep me posted. Honestly if you get it working let me know what steps were required, would be great to add info the setup guide!

**Comment by lowfuel** [comment]
when you launch prd does it say it's using "cpu" or "cuda:0" (or similar?)
I don't know if ROCm basically pretends to be a CUDA device, or if you have to add support in the code for another type of device (for example on the Apple M1 I have to change it from "cpu" to "mps". I feel like I read somewhere that ROCm needs it set to "hip" but I can't recall for sure.

**Comment by lowfuel** [comment]
Correct!

**Comment by lowfuel** [comment]
This seems to imply you should use "cuda":

https://pytorch.org/docs/stable/notes/hip.html

**Comment by Julianoe** [comment]
According to the tests in[ this doc page ](https://pytorch.org/get-started/locally/#linux-verification) the first verification succeeds but then if I run the following code it returns nothing, indicating there is something wrong with my "your GPU driver and CUDA being enabled and accessible by PyTorch"

```
import torch
torch.cuda.is_available()
```

I'm continuing my research

**Comment by Julianoe** [comment]
I'm almost there. I reinstalled on a new fresh environment, ROCm still works, all packages are good, versions of Pytorch seem to be good too, when I execute `python3` to enter the python command and it does not return FALSE:
```
>> import torch;
>>> torch.cuda.is_available;
<function is_available at 0x7ffbf73bcd40>
>>> 
```

My only problem is that once in the script, it does not detect any CUDA GPUS when I put 'cuda' although I added the command `print(torch.cuda.is_available);` at line 1027 and it outputs the same as above, suggesting, I suppose It works in a way. It just lacks a little something to detect use the GPU properly I guess.

```
RuntimeError: No CUDA GPUs are available
```
